src/com/google/enterprise/adaptor/database/UniqueKey.java

```diff
@@ -75,22 +77,22 @@
       log.fine("element: `" + e + "'");
       e = e.trim();
       String def[] = e.split(":", 2);
-      if (2 != def.length) {
-        String errmsg = "Invalid UniqueKey definition for '" + e + "'. Valid"
-            + " definition is 'column_name:type' where types can be int, string,"
-            + " timestamp, date, time and long.";
-        throw new InvalidConfigurationException(errmsg);
-      }
-      def[0] = def[0].trim();
-      def[1] = def[1].trim();
-      String name = def[0];
+      String name = def[0].trim();
       ColumnType type;
-      try {
-        type = ColumnType.valueOf(def[1].toUpperCase(US));
-      } catch (IllegalArgumentException iae) {
-        String errmsg = "Invalid UniqueKey type '" + def[1] + "'. Valid"
-            + " types are: " + ColumnType.values();
-        throw new InvalidConfigurationException(errmsg);
+
+      if (def.length == 1) {
+        // No type given. Try to get type from ResultSetMetaData later.
```

Comment:
Only some confusion about uniqueKey vs metadataColumns. I've seen config files that had things like: db.metadataColumns=thiscolumn:int, thatcolumn:string.

More generally, people have lots of problems configuring the database connector, and we've been looking at the bug reports, support cases, and also conversations with QA and support. Beyond those specifics, we've been trying to make it both easier to configure in whatever ways we could (e.g., allowing the identity mapping by default in db.metadataColumns), and have the connector be more specific about what's wrong when there are config errors.

Back to the specifics, this is a case where the database knows the data we need, and we now even look at the data we need in init, so it seems a little silly not to use the data to simplify the configuration.